kong/runloop/balancer.lua

```diff
@@ -921,7 +974,15 @@ local function init()
 
   if kong.configuration.worker_consistency == "eventual" then
```

Comment:
This is a bit unrelated, but is there reason that timers are only enabled on eventual? I mean that with `router` and `plugins` iterator we do eventual also on `strict` mode. The difference is that on `strict` we have stricter check of version on request time (e.g. if it is not the latest we will try harder to build a new one).

Reply:
Yes, I guess we can make `strict` better in that regard. But it's out of scope from this PR, here we're trying to just improve the db performance on `eventual` only. On `strict` the db performance is very bad for users that have a big amount of upstreams in the database and/or users that make frequent upstream Admin API updates.

Biggest difference between `strict` and `eventual` on the balancer side is, AFAIK, here: https://github.com/Kong/kong/blob/next/kong/runloop/balancer.lua#L630-L646

On strict, if we receive an upstream event, we will load all the upstreams from the DB when a request comes in (if we don't receive an upstream event then all the upstreams will be in the cache already). On eventual, we will just use what’s available in the cache when a request comes in and the timer will eventually load all the upstreams from the database.

Reply:
Some db load tests will be really needed here on this PR.